hnn_core/dipole.py

```diff
@@ -53,8 +57,27 @@ def simulate_dipole(net, n_trials=None, record_vsoma=False,
 
     # XXX needed in mpi_child.py:run()#L103; include fix in #211 or later PR
     net._params['N_trials'] = n_trials
-    net._instantiate_drives(n_trials=n_trials)
-    net.cell_response.reset()  # see #290 for context; relevant for MPI
+
+    # XXX used in network_builder::_simulate_single_trial
+    # and network_builder.__init__()
+    net._params['tstop'] = tstop
+    net._params['dt'] = dt
```

Comment:
Are we planning on keeping `tstop` and `dt` as elements of the param dict in the future?

Reply:
Why not just pass `tstop` and `dt` to `_simulate_single_trial()`? Maybe it's time to create a `simulation_status` dict that gets passed around `_clone_and_simulation()` and `_simulate_single_trial()` that contains some of the relevant simulation arguments including the `trial_idx`.

Reply:
I'd love that. It's just that it gets really hairy with the MPI backend ... see my first attempt in #315 . I would love to fix this but I think it would be a whole different PR. You'd need to pass `tstop` to from `MPIBackend.simulate` to `mpi_child.py` where it gets unpickled and where it's used by `_clone_and_simulate`. I was hitting segmentation faults and decided it was not worth the pain

Reply:
Oh I see it now. We can certainly save this for a later PR.